test/functional/outputs/splunk/forward_to_splunk_test.go

```diff
@@ -194,4 +216,219 @@ var _ = Describe("Forwarding to Splunk", func() {
+			}
+			Expect(framework.Deploy()).To(BeNil())
+
+			// Wait for splunk to be ready
+			waitOnSplunk()
+
+			if inputType == obs.InputTypeApplication {
+				if expSource == "" {
+					expSource = strings.Join([]string{framework.Namespace, framework.Pod.Name, "collector"}, "_")
+				}
+				// Write app logs
+				timestamp := "2020-11-04T18:13:59.061892+00:00"
+				applicationLogLine := functional.NewCRIOLogMessage(timestamp, "This is my test message", false)
+				Expect(framework.WriteMessagesToApplicationLog(applicationLogLine, 2)).To(BeNil())
+			}
+
+			if inputType == obs.InputTypeAudit {
+				if expSource == "" {
+					expSource = "auditd"
+				}
+				// Write audit logs
+				timestamp, _ := time.Parse(time.RFC3339Nano, "2024-04-16T09:46:19.116+00:00")
+				auditLogLine := functional.NewAuditHostLog(timestamp)
+				writeAuditLogs := framework.WriteMessagesToAuditLog(auditLogLine, 1)
+				Expect(writeAuditLogs).To(BeNil(), "Expect no errors writing audit logs")
+			}
+
+			if inputType == obs.InputTypeInfrastructure {
+				if expSource == "" {
+					expSource = "google-chrome.desktop"
+				}
+				// Write journal logs
+				logline := functional.NewJournalLog(0, "journal message", "functional")
+				Expect(framework.WriteMessagesToInfraJournalLog(logline, 1)).To(BeNil())
+			}
+
+			// Read logs
+			logs, err := framework.ReadLogsByTypeFromSplunk(framework.Namespace, framework.Name, string(inputType))
+			Expect(err).To(BeNil(), "Expected no errors getting logs from splunk")
+			Expect(logs).ToNot(BeEmpty())
+
+			result, err := framework.ReadFieldByIndexFromSplunk(framework.Namespace, framework.Name, "main", "source", "json")
+			Expect(err).To(BeNil(), "Expected no errors getting field from splunk")
+			for _, v := range result {
+				matches := regexpSource.FindStringSubmatch(v)
+				Expect(matches[1]).To(Equal(expSource), "Expected to find match for source")
+			}
+
+		},
+			Entry("should send application logs with dynamic source field", `{.log_type||"missing"}`, "application", obs.InputTypeApplication),
+			Entry("should send application logs with static + dynamic source field", `foo-{.log_type||"missing"}`, "foo-application", obs.InputTypeApplication),
+			Entry("should send application logs with static + label with dot/slash source field", `foo-{.kubernetes.labels."slash/test.dot"||"missing"}`, "foo-application", obs.InputTypeApplication),
+			Entry("should send application logs with static + fallback value's source field", `foo-{.missing||"application"}`, "foo-application", obs.InputTypeApplication),
+			Entry("should send application logs with default source ", "", "", obs.InputTypeApplication),
+			Entry("should send audit logs with default source ", "", "", obs.InputTypeAudit),
+			Entry("should send journal logs with default source ", "", "", obs.InputTypeInfrastructure))
+
+		DescribeTable("with user defined payloadKey", func(payloadKey, expSourceType string, expression *regexp.Regexp) {
```

Comment:
Please add a test for the audit log defined in the feature request and see how closely we can get to matching the request.  I don't believe we wish to test indexing every annotation but I'm interested to see how we populate "event" with the audit event message and maybe index a few other fields